tests/Integration/RestTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Parsely REST API tests.
+ *
+ * @package Parsely\Tests
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Tests\Integration;
+
+use Parsely\Parsely;
+use Parsely\Rest;
+
+
+/**
+ * Parsely REST API tests.
+ */
+final class RestTest extends TestCase {
+	/**
+	 * Internal variable
+	 *
+	 * @var Rest $rest Holds the Rest object
+	 */
+	private static $rest;
+
+	/**
+	 * Internal Parsely variable
+	 *
+	 * @var Parsely $parsely Holds the Parsely object
+	 */
+	private static $parsely;
+
+	/**
+	 * The setUp run before each test
+	 */
+	public function set_up(): void {
+		parent::set_up();
+
+		self::$parsely = new Parsely();
+		self::$rest    = new Rest( self::$parsely );
+	}
+
+	/**
+	 * Test whether the logic has been enqueued in the `rest_api_init` hook.
+	 *
+	 * @covers \Parsely\Rest::run
+	 */
+	public function test_register_enqueued_rest_init(): void {
+		self::$rest->run();
+		self::assertSame(
+			10,
+			has_action( 'rest_api_init', array( self::$rest, 'register_parsely_meta' ) )
+		);
+	}
+
+	/**
+	 * Test whether the logic has been enqueued in the `rest_api_init` hook with a filter that disables it.
+	 *
+	 * @covers \Parsely\Rest::run
+	 */
+	public function test_register_enqueued_rest_init_filter(): void {
+		add_filter( 'wp_parsely_enable_rest_api_support', '__return_false' );
+		self::$rest->run();
+		self::assertFalse( has_action( 'rest_api_init', array( self::$rest, 'register_parsely_meta' ) ) );
+	}
+
+	/**
+	 * Test that the REST fields are registered to WordPress REST API.
+	 *
+	 * @covers \Parsely\Rest::register_parsely_meta
+	 */
+	public function test_register_parsely_meta_registers_fields(): void {
+		global $wp_rest_additional_fields;
+
+		self::$rest->register_parsely_meta();
+
+		$this->assertParselyRestFieldIsConstructedCorrectly( 'page', $wp_rest_additional_fields );
+
+		// Cleaning up the registered fields.
+		// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedVariableFound
+		$wp_rest_additional_fields = array();
+	}
+
+	/**
+	 * Test that the REST fields are can be modified using the `wp_parsely_rest_object_types` filter.
+	 *
+	 * @covers \Parsely\Rest::register_parsely_meta
+	 */
+	public function test_register_parsely_meta_with_filter(): void {
+		global $wp_rest_additional_fields;
+
+		add_filter(
+			'wp_parsely_rest_object_types',
+			function( $object_types ) {
```

Comment:
```suggestion
			function() {
```

per: https://sonarcloud.io/project/issues?id=Parsely_wp-parsely&issues=AX2ftYPDuw8Yrhl6Lsoo&open=AX2ftYPDuw8Yrhl6Lsoo&pullRequest=489